pageserver/src/tenant/upload_queue.rs

```diff
@@ -56,6 +57,9 @@ pub(crate) struct UploadQueueInitialized {
     /// DANGER: do not return to outside world, e.g., safekeepers.
     pub(crate) latest_metadata: TimelineMetadata,
 
+    /// Part of the flattened "next" `index_part.json`.
+    pub(crate) latest_lineage: Lineage,
```

Comment:
In some future PR: I will look into having just a `dirty_index_part` here instead of these separate things.

Then we could just not copy anything, only serialize the index_part into bytes when scheduling an upload while also pre-calculating the *post-upload* remote size and be done with it.

Rationale: I think this is quite confusing that the fields are separate like this, and that the IndexPart is serialized again for each upload attempt.